File: 参考/YCrispBiscuit-BestFriend/Server/app/services/health_service.py
from app.schemas.health_schema import HealthResponse
from app.mappers.health_mapper import HealthMapper


class HealthService:
    """业务逻辑层：每文件一个类，依赖注入 mapper"""

    def __init__(self, mapper: HealthMapper | None = None) -> None:
        self._mapper = mapper or HealthMapper()

    async def get_health(self) -> HealthResponse:
        try:
            data = await self._mapper.select_health_raw()
            return HealthResponse(**data)
        except Exception as e:
            raise ValueError(f"Failed to get health: {str(e)}")

File: 参考/YanJing-Manager/RegionAI-Node-Server/YanJing-Node-Server/src/services/HealthService.ts
import { HealthMapper } from '../mappers/HealthMapper.js'
import type { HealthResponseSchema } from '../schemas/HealthSchema.js'

export class HealthService {
  private readonly healthMapper: HealthMapper

  constructor(mapper?: HealthMapper) {
    this.healthMapper = mapper || new HealthMapper()
  }

  async getHealth(): Promise<HealthResponseSchema> {
    try {
      const data = await this.healthMapper.selectHealthRaw()
      return data // 对应 FastAPI 中的 HealthResponse(**data)
    } catch (e: unknown) {
      const errorMessage = e instanceof Error ? e.message : String(e)
      throw new Error(`Failed to get health: ${errorMessage}`, { cause: e })
    }
  }
}
